R/utils-numeric.R

```diff
@@ -1,3 +1,66 @@
+## TODO think about the relationship between this and the plot.data bin fxns
+#' Find Bin Ranges
+#' 
+#' This function will find bin start, end and labels for a 
+#' continuous variable. Optionally, it can return a value/ count
+#' per bin. By default returns 10 bins for equalRanges and quantile
+#' methods and 6 for sd (standard deviations).
+#' @param x Numeric (or Date) vector to find bins for
+#' @param method A string indicating which method to use to find bins ('equalRanges', 'quantile', 'sd')
+#' @param numBins A number indicating how many bins are desired
+#' @param getValue A boolean indicating whether to return the counts per bin
+#' @export 
+getBinRanges <- function(x, method = c('equalInterval', 'quantile', 'sd'), numBins = NULL, getValue = c(TRUE, FALSE)) {
+  method <- veupathUtils::matchArg(method)
+  getValue <- veupathUtils::matchArg(getValue)
+  if (is.null(numBins)) numBins <- 10
+
+  isDate <- FALSE
+  if (class(x) == 'Date') {
+    x <- as.numeric(x)
+    isDate <- TRUE
+  }
+
+  binEdges <- unname(breaks(x, method, numBins))
+  if (anyDuplicated(binEdges)) {
+    warning("There is insufficient data to produce the requested number of bins. Returning as many bins as possible.")
+    binEdges <- unique(binEdges)
+  }
+
+  binStart <- binEdges[1:(length(binEdges)-1)]
+  binEnd <- binEdges[2:length(binEdges)]
+  if (isDate) {
+    binStart <- as.Date(binStart, origin = "1900-01-01") # this is the default origin
+    binEnd <- as.Date(binEnd, origin = "1900-01-01")
+  } else {
+    # do we want formatC here? itll give us strings, but theyll be pretty
+    # alternative is maybe to round to some number of digits or signifs
+    binStart <- formatC(binStart)
+    binEnd <- formatC(binEnd)
```

Comment:
@bobular and i had a really fun time working with this rounding bins bit last july :) 
i think we ended up rounding to some number of sigfigs, but we had to do the rounding before breaks, and then also make sure that all the data points were included in the new bins.
If the need is for speedy deploys, we could just leave it with formatC and then we can revisit what we want to do later. It's a problem for bins everywhere :D

Reply:
looks like our adventures are still present in the breaks function :) 
Since the bin edges are coming out of `breaks`, they should be pretty nice looking already. Modifying the bin start or end out here would make it possible to round the bin start to a number greater than the min value of the dataset

Reply:
maybe we could just move that logic to its own function? that way all of our bin rounding would be consistent

Reply:
im maintaining the decision to not touch the cut* fxns in this PR too much. but i am taking a page from their book. im going to only format the labels, and leave the binStart and binEnds exact values.